### PR TITLE
Tweaks Resource Management in Space, and adds more ruins

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Increases the resources allocated for space, space ruins now have different price costs depending on complexity.
Adds a bunch more ruins.

## Why It's Good For The Game

Space has been neglected for quite while. It's time to add more ruins and revise the ruin generation code.

Made a new PR cause I merged the wrong branch into it and reversing only made things worse.

## Changelog
:cl:
add: Changed the memory allocation for space ruin generation
add: Added new ruins
/:cl:

### CODE REVIEW

File: code/controllers/configuration/entries/game_options.dm [hunk 1]
```diff
@@ -372,7 +372,7 @@
 	min_val = 0
 
 /datum/config_entry/number/space_budget
-	config_entry_value = 16
+	config_entry_value = 60
```
Comment (PowerfulBacon) [inline code/controllers/configuration/entries/game_options.dm]: 60 ruins

Comment (eeSPee) [reply code/controllers/configuration/entries/game_options.dm]: I change the cost value of ruins.

Unless it doesn't account the cost at all.

Comment (PowerfulBacon) [reply code/controllers/configuration/entries/game_options.dm]: Pretty sure all ruins count as 1 as found in bluespace exploration

Comment (PowerfulBacon) [reply code/controllers/configuration/entries/game_options.dm]: ah you changed the cost, check its actually used

Comment (eeSPee) [reply code/controllers/configuration/entries/game_options.dm:375]: Ok. Also, why won't it compile?

**Comment by St0rmC4st3r** [comment]
@eeSPee The maps are in the wrong format. Convert to TGM.

**Comment by eeSPee** [comment]
> @eeSPee The maps are in the wrong format. Convert to TGM.

I have no idea how to do that.

**Comment by Archanial** [comment]
`ERROR: Non-TGM formatted map detected. Please convert it using Map Merger!`

use map merger, it's in tools/mapmerge2/mapmerge.bat

File: code/datums/ruins/space.dm [hunk 2]
```diff
@@ -254,61 +364,106 @@
...
+	cost = 1
+
+/datum/spacedock13/ruin/space/swarmerstation
+	id = "swarmerstation13"
+	suffix = "swarmerstation13.dmm"
+	name = "Swarmer Station"
+	description = "Shat is left of a syndicate radio station, after swarmers infiltrated it."	
+	cost = 2
+
+/datum/spacedock13/ruin/space/refference
```
Comment (park66665) [inline code/datums/ruins/space.dm]:
```suggestion
/datum/map_template/ruin/space/spacedock
	id = "spacearcade"
	suffix = "spacedock13.dmm"
	name = "Shuttle Construction"
	description = "A shuttle building station that seems to have been forgotten, or floated off into space."	
	cost = 2

/datum/map_template/ruin/space/spacehive
	id = "spacehive"
	suffix = "spacehive.dmm"
	name = "Space Hive"
	description = "A hydroponics experiment gone horribly, horribly wrong!"
	cost = 1

/datum/map_template/ruin/space/swarmerstation
	id = "swarmerstation13"
	suffix = "swarmerstation13.dmm"
	name = "Swarmer Station"
	description = "Shat is left of a syndicate radio station, after swarmers infiltrated it."	
	cost = 2

/datum/map_template/ruin/space/refference
```

Come on, I know you are better than *this*.

Comment (eeSPee) [reply code/datums/ruins/space.dm]: :)

File: code/controllers/configuration/entries/game_options.dm (hunk 1 above)
Comment (eeSPee) [reply code/controllers/configuration/entries/game_options.dm]: I checked, and they use the same code as lavaland ruins, so cost is taken into accound.

It was just that all cost was at 1.

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 3]
```diff
@@ -304,3 +304,11 @@
 /mob/living/simple_animal/hostile/poison/bees/short/Initialize()
 	. = ..()
 	addtimer(CALLBACK(src, .proc/death), 50 SECONDS)
+
+
+/mob/living/simple_animal/hostile/poison/bees/space
+	name = "killer space bee"
+	desc = "I mean, killer is 'IN' the name..."
```
Comment (utilbird) [inline code/modules/mob/living/simple_animal/hostile/bees.dm:311]:
```suggestion
	desc = "I mean, killer is \'IN\' the name..."
```

Comment (utilbird) [reply code/modules/mob/living/simple_animal/hostile/bees.dm:311]: Oh wait, nevermind lol

File: _maps/RandomRuins/SpaceRuins/spacearcade.dmm [hunk 4]
```diff
@@ -0,0 +1,658 @@
...
+/turf/open/floor/light/colour_cycle/dancefloor_a,
+/area/ruin/space/has_grav/powered)
+"f" = (
+/turf/open/floor/light/colour_cycle/dancefloor_b,
+/area/ruin/space/has_grav/powered)
+"g" = (
+/obj/machinery/computer{
+	name = "Dance Dance Revolution";
+	desc = "Unfortunately, the machine seems to be broken..."
+	},
```
Comment (park66665) [inline _maps/RandomRuins/SpaceRuins/spacearcade.dmm]: https://en.wikipedia.org/wiki/Dance_Dance_Revolution

File: code/datums/ruins/space.dm [hunk 5]
```diff
@@ -254,61 +364,106 @@
...
-	You will have to power areas to raise the bolts on the doors. look out for secrets."
\ No newline at end of file
+	You will have to power areas to raise the bolts on the doors. look out for secrets."
+	cost = 4
+
+/datum/map_template/ruin/space/spacearcade
+	id = "spacearcade"
+	suffix = "spacearcade.dmm"
+	name = "Space Arcade"
+	description = "Come here to have fun! We even have the best arcade game in the Universe, Dance Dance Revolution!"
```
Comment (park66665) [inline code/datums/ruins/space.dm]: ditto.

**Review by park66665 — COMMENTED** [review]
1. refference.dmm, spacehive.dmm need to replace their aired version of turfs to airless version of it, and if there isn't, you should create one.
2. sos series of maps needs more tiny fans.
3. You should use `pixel_x = 32` instead of `pixel_x = 24`.

Screenshot 2021-03-06 221733

File: _maps/RandomRuins/SpaceRuins/spacearcade.dmm (hunk 4 above)
Comment (eeSPee) [reply _maps/RandomRuins/SpaceRuins/spacearcade.dmm]: Right

Comment (park66665) [reply _maps/RandomRuins/SpaceRuins/spacearcade.dmm]: You probably shouldn't use that name directly. Change the name slightly.

**Comment by eeSPee** [comment]
Thanks for the feedback. That should be resolved.

**Comment by wraith-54321** [comment]
this will work very well if the shuttle rework gets merged/finished

**Comment by eeSPee** [comment]
Should I break it down and remove the space resource management, and just add the ruins?

File: code/controllers/configuration/entries/game_options.dm (hunk 1 above)
Comment (ike709) [inline code/controllers/configuration/entries/game_options.dm]:
```suggestion
	config_entry_value = 40
```

Maploading is still expensive even if it doesn't OOM us. Tweak the value and I'll merge.

**Review by ike709 — APPROVED** [review]

**Pull request merged**